File: ClassLibrary1/Entity/Question.cs

namespace ClassLibrary1.Entity
{
    public class Question
    {
        public int Id { get; set; }
        public int QuestionNumber { get; set; } = 1;
        public string Context { get; set; } = "";
        public string Heading { get; set; } = "";
        public string Text { get; set; } = "";
        public ICollection<Option> Options { get; set; } = new List<Option>();
        public Survey? Survey { get; set; }

        // this relationship is configured as a required, since the Question.SurveyId foreign key property is non-nullable
        // https://docs.microsoft.com/en-us/ef/core/saving/cascade-delete#deleting-a-principalparent
        public Guid SurveyId { get; set; }      

        public override string ToString()
        {
            return Id + "  " + Text;
        }
    }
}

File: reactapp1/src/model/Question.ts
import { Option } from "./Option";

export class Question {
    id:string='';
    questionNumber?: number;
    heading?: string;
    text?: string;    
    options?: Option[] = [];

    constructor(questionFormValues: QuestionFormValues, questionNumber: number) {
        console.log(questionFormValues.option1);
        this.id='0';
        this.questionNumber = questionNumber;
        this.heading = questionFormValues.heading;
        this.text = questionFormValues.text;
        if (questionFormValues?.option1)
            this.options?.push(new Option(questionFormValues.option1))
        if (questionFormValues?.option2)
            this.options?.push(new Option(questionFormValues.option2))
        if (questionFormValues?.option3)
            this.options?.push(new Option(questionFormValues.option3))
        if (questionFormValues?.option4)
            this.options?.push(new Option(questionFormValues.option4))
        if (questionFormValues?.option5)
            this.options?.push(new Option(questionFormValues.option5))
        if (questionFormValues?.option6)
            this.options?.push(new Option(questionFormValues.option6))
    }
}


export class QuestionFormValues {
    heading?: string = '';
    text?: string = '';
    option1?: string = '';
    option2?: string = '';
    option3?: string = '';
    option4?: string = '';
    option5?: string = '';
    option6?: string = '';
}